### ISSUE TITLE
not every ASPP parameter included in `param_groups`

### ISSUE THREAD

**Issue by PortikAbel** [PortikAbel/ReProSeg#12]
`_init_param_groups()` collects parameters from `self.layers.backbone` and only a single ASPP conv weight (`self.layers.shared_weights`) for the head. The remaining parameters in `self.layers.aspp_convs` (other conv weights + BatchNorm params) are not included in any optimizer, so they will never be trained. Consider adding `self.layers.aspp_convs.parameters()` (or a dedicated head module) to the appropriate param group.
```suggestion
        self.param_groups["classifier_head"].append(self.layers.shared_weights)
        for _name, param in self.layers.aspp_convs.named_parameters():
            if param is not self.layers.shared_weights:
                self.param_groups["classifier_head"].append(param)
```

_Originally posted by @Copilot in https://github.com/PortikAbel/ReProSeg/pull/9#discussion_r3080454922_
